src/styles/components/main.scss

```diff
@@ -0,0 +1,26 @@
+.main {
+  display: grid;
+  padding: 50px 40px;
+  grid-template-columns: repeat(4, 200px);
+  gap: 46px 48px;
+  justify-content: center;
+  box-sizing: border-box;
+}
+
+@media (max-width: 1023px) {
+  .main {
+    grid-template-columns: repeat(3, 200px);
```

Comment:
Similarly, replace the fixed `200px` with `1fr` in the media query to maintain a flexible layout on smaller screens. For instance, `grid-template-columns: repeat(3, 1fr);`.